src/core/lint-hrefs-ids.js

```diff
@@ -0,0 +1,21 @@
+// Module core/lint-hrefs-ids
+// Warns about href's that do not have corresponding id's
+
+import { pub } from "core/pubsubhub";
+
+export const name = "core/lint-hrefs-ids";
+
+export function run(conf, doc, cb) {
+  console.log()
+  Array.from(doc.querySelectorAll('a[href]'))
+    .map(a => ({ a, href: a.getAttribute('href') }))
+    .filter(({ href }) => Boolean(href))
+    .filter(({ href }) => href.startsWith('#'))
+    .map(({ a, href }) => ({ a, href, id: href.slice(1) }))
```

Comment:
You can probably get away with doing away with this one too... just keep reusing the `a` without mapping it (or destructure it as you go, but you don't need to map it).